TypeSystem/flow-analysis/promotion_via_assignment_A03_t06.dart

```diff
@@ -0,0 +1,26 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion We say that a variable `x` is promotable via assignment of an
+/// expression of type `T` given variable model `VM` if
+/// - `VM = VariableModel(declared, promoted, tested, assigned, unassigned,
+///     captured)`
+/// - and captured is false
+/// - and `S` is the current type of `x` in `VM`
+/// - and `T <: S` and not `S <: T`
+/// - and `T` is a type of interest for `x` in `tested`
+///
+/// @description Checks that if `T` is not subtype of `S` then promotion via
+/// assignment cannot be performed. Test type `Object?`;
+/// @author sgrekhov22@gmail.com
+/// @issue 60620
+
+import '../../Utils/static_type_helper.dart';
+
+main() {
+  String s = "x";
+  if (s is Object?) {} // ignore: unnecessary_type_check
+  s = "y"; // Cannot assign `Object?`, let's assign `String <: Object?`
```

Comment:
This is again not really doing what we need. `s = "y"` is just an assignment of an expression whose type is the declared type of the variable which hasn't been promoted. There's simply nothing going on about the type of `s` here, and there is certainly no reason why we would try to "promote" `s` to have type `Object?`.

Perhaps this test is just impossible? ;-)

Reply:
Updated. `test2` removed, `test1()` renamed to `main()`